### ISSUE TITLE
Allow PHPStorm Code Coverage integration by providing `/opt/phpstorm-coverage` in web container

### ISSUE THREAD

**Issue by nico-loeber** [drud/ddev#3920]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe

PHPStorm provides you with the ability to run your PHP unit tests with coverage report directly from the GUI. After execution, you can see in your IDE which lines are covered and which are not.

Sadly this does fail currently in DDEV with the error message 
```
Generating code coverage report in Clover XML format ... failed [00:00.345]
Directory "/opt/phpstorm-coverage" could not be created
```

The path is auto generated by PHPStorm, for example `--coverage-clover /opt/phpstorm-coverage/acol@UdpConnectionTest_testConnect.xml` and I didn't find an option to change the prefix of the generated path.

### Describe the solution you'd like

You can manually create this directory and give it write permission for your user so PHPStorm can create the coverage reports in there but it would be great if this directory would already exist with appropriate permissions, so you can use this PHPStorm feature with zero configuration.

In this issue most of the people add this directory as a docker volume.
https://intellij-support.jetbrains.com/hc/en-us/community/posts/360006971460-Phpstorm-code-coverage-with-docker-container-
I am not aware of the advantges / disadvantages of mounting a docker volume instead of just providing the directory in the container for the coverage reports. I guess it would make the coverage reports persistent when restarting ddev? Your feedback is welcome.

So the solution I would like would be adding a docker volume mount or a directory at /opt/phpstorm-coverage with write permissions for the ddev user to the web container.

### Describe alternatives you've considered

The alternative is adding a DDEV post-start hook executing:

```bash
mkdir -p "/opt/phpstorm-coverage" && chmod a+rw "/opt/phpstorm-coverage"
```

### Additional context

In combination with https://github.com/drud/ddev/pull/3754 and the [DDEV Integration Plugin](https://github.com/php-perfect/ddev-intellij-plugin) solving this issue will allow the use of all test run features (Run, Debug, Run with Coverage) of PHPStorm with zero configuration.

**Comment by rfay** [comment]
You have 3 choices:
1. Bind-mount to something like .ddev/code-coverage or something (persists)
2. Docker volume (not available on the host side but persists)
3. Just create the directory in the image and it doesn't persist. (`ddev restart` throws it all away)

If it persistence doesn't matter the third option works. 

You could probably put it in the ddev-global-cache (indexed by project name)

But... Isn't PhpStorm accessing this from the host side? If so, a docker volume isn't very useful, and only a bind mount would be useful. But where is the code coverage being created? Inside the container or on the host? Are you sure "Directory "/opt/phpstorm-coverage" could not be created" isn't complaining about something happening on the the host side?

**Comment by nico-loeber** [comment]
> But... Isn't PhpStorm accessing this from the host side? If so, a docker volume isn't very useful, and only a bind mount would be useful. But where is the code coverage being created? Inside the container or on the host? Are you sure "Directory "/opt/phpstorm-coverage" could not be created" isn't complaining about something happening on the the host side?

Yes, creating the directory inside the container fixes the issue. PHPStorm is generating and accessing the code coverage results inside the container.

After your explaination I would go with 2. Docker volume from the perspective of a PHPStorm user who likes to use this feature.
Is there much overhead when creating a docker volume? Guess not every ddev user is a PHPStorm users and only a few will use this feature.

**Comment by rfay** [comment]
The docker volume doesn't cost much, it's the fastest and most efficient way to have something in there. You could just link the directory it wants to /mnt/ddev-global-config/phpstorm/<project>/coverage, and you wouldn''t have to create or maintain a volume. The only issue then is that directory has to get created sometime (/mnt/ddev-global-config/...)

**Comment by nico-loeber** [comment]
Thanks for your feedback!
I will try to create a pull request for that.

**Comment by rfay** [comment]
This was fixed in #3924
